File: iosApp/iosAppUITests/NearbyLineView.swift
//
//  NearbyLineView.swift
//  iosApp
//
//  Created by Simon, Emma on 6/18/24.
//  Copyright © 2024 MBTA. All rights reserved.
//

import shared
import SwiftUI

struct NearbyLineView: View {
    let nearbyLine: StopsAssociated.WithLine
    let pinned: Bool
    let onPin: (String) -> Void
    let pushNavEntry: (SheetNavigationStackEntry) -> Void
    let now: Instant

    var body: some View {
        LineCard(line: nearbyLine.line, routes: nearbyLine.routes, pinned: pinned, onPin: onPin) {
            ForEach(Array(nearbyLine.patternsByStop.enumerated()), id: \.element.stop.id) { index, patternsAtStop in
                VStack(spacing: 0) {
                    NearbyStopView(
                        patternsAtStop: patternsAtStop,
                        condenseHeadsignPredictions: nearbyLine.condensePredictions,
                        now: now,
                        pushNavEntry: pushNavEntry,
                        pinned: pinned
                    )
                    if index < nearbyLine.patternsByStop.count - 1 {
                        Divider().background(Color.halo)
                    }
                }
            }
        }
    }
}

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/nearbyTransit/NearbyLineView.kt
package com.mbta.tid.mbta_app.android.nearbyTransit

import androidx.compose.runtime.Composable
import com.mbta.tid.mbta_app.android.component.legacyRouteCard.LineCard
import com.mbta.tid.mbta_app.model.StopDetailsFilter
import com.mbta.tid.mbta_app.model.StopsAssociated
import kotlinx.datetime.Instant

@Composable
fun NearbyLineView(
    nearbyLine: StopsAssociated.WithLine,
    pinned: Boolean,
    onPin: (String) -> Unit,
    now: Instant,
    onOpenStopDetails: (String, StopDetailsFilter?) -> Unit,
    showElevatorAccessibility: Boolean = false
) {
    LineCard(nearbyLine.line, nearbyLine.routes, pinned, onPin) {
        for (patternsAtStop in nearbyLine.patternsByStop) {
            NearbyStopView(
                patternsAtStop,
                condenseHeadsignPredictions = nearbyLine.condensePredictions,
                now,
                pinned,
                onOpenStopDetails,
                showElevatorAccessibility
            )
        }
    }
}
